skops/hub_utils/_hf_hub.py

```diff
@@ -205,14 +234,41 @@ def recursively_default_dict() -> MutableMapping:
         config["sklearn"]["example_input"] = _get_example_input(data)
         config["sklearn"]["columns"] = _get_column_names(data)
     elif "text" in task:
-        if isinstance(data, list) and all(isinstance(x, str) for x in data):
-            config["sklearn"]["example_input"] = {"data": data[:3]}
+        if _is_iterable_of_strings(data):
+            config["sklearn"]["example_input"] = {
+                "data": list(itertools.islice(data, 3))
+            }
         else:
-            raise ValueError("The data needs to be a list of strings.")
+            raise ValueError("The data needs to be an iterable of strings.")
 
     dump_json(Path(dst) / "config.json", config)
 
 
+def _is_iterable_of_strings(data):
+    """Checks whether data is an iterable of strings.
+
+    Parameters
+    ----------
+    data: Any
+        Any object.
+
+    Returns
+    -------
+    is_iterable_of_strings: bool
+        A boolean variable indicating whether or not data is an iterable of
+        strings.
+    """
+    if isinstance(data, str):
+        return False
+    try:
+        # needed in case data is an iterator or a generator
+        data, data_copy = itertools.tee(data, 2)
+        return all(isinstance(x, str) for x in data_copy)
```

Comment:
I'm undecided on this one. Here we're testing all the data, even though we only need the first 3 samples. Of course, we can argue that fitting the model should not work if some rows have strange types, but is it really our job here to verify this?